cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractOperatorState.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.strimzi.api.kafka.model.KafkaRebalance;
+import io.strimzi.api.kafka.model.status.Condition;
+import io.strimzi.api.kafka.operator.assembly.KafkaRebalanceState;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+
+public class AbstractOperatorState {
+
+
+    public static Matcher<KafkaRebalance> hasState(final KafkaRebalanceState state) {
+        return new TypeSafeDiagnosingMatcher<>() {
+
+            @Override
+            protected boolean matchesSafely(KafkaRebalance kafkaRebalance, Description mismatchDescription) {
+                mismatchDescription.appendText("was ").appendValue(kafkaRebalance);
+
+                if (kafkaRebalance == null) {
+                    mismatchDescription.appendText("\n KafkaRebalance is null");
+                    return false;
+                }
+
+                if (kafkaRebalance.getStatus() == null) {
+                    mismatchDescription.appendText("\n KafkaRebalance status is null");
+                    return false;
+                }
+
+                if (kafkaRebalance.getStatus().getConditions() == null) {
+                    mismatchDescription.appendText("\n KafkaRebalance status conditions is null");
+                    return false;
+                }
+
+                return true;
+            }
+
+            @Override
+            public void describeTo(final Description description) {
+                description.appendText("Expected state is ").appendValue(state);
+            }
+        };
+
+    }
+
+
+    /**
+     * Checks all conditions in the supplied status to see if type of one of them matches the supplied rebalance state.
+     * @param state he expected rebalance state to be searched for.
+     * @return
+     */
+    public static Matcher<List<Condition>> hasStateInConditions(KafkaRebalanceState state) {
+        return new TypeSafeDiagnosingMatcher<>() {
+
+            @Override
+            protected boolean matchesSafely(List<Condition> conditions, Description mismatchDescription) {
+                mismatchDescription.appendText("was ").appendValue(conditions);
+
+                List<String> foundStatuses = new ArrayList<>();
+
+                for (Condition condition :  conditions) {
```

Comment:
if conditions is null, we are going to have a bad time.
Worth a check, admittedly this is test code and it should all blow up anyway if it was null, but better to have a kroe descriptive usage error

Reply:
Right, I can remove the condition null check, but maybe in this case we don't need also this part (in `Matcher<KafkaRebalance> hasState()` method)
`                if (kafkaRebalance == null) {
                    mismatchDescription.appendText("\n KafkaRebalance is null");
                    return false;
                }

                if (kafkaRebalance.getStatus() == null) {
                    mismatchDescription.appendText("\n KafkaRebalance status is null");
                    return false;
                }

                if (kafkaRebalance.getStatus().getConditions() == null) {
                    mismatchDescription.appendText("\n KafkaRebalance status conditions is null");
                    return false;
                }`
Because in the end we are trying to check if the `conditions` is not null. WDYT ?

Reply:
I would put this logic in hasState()

And then call hasState() in hasStateInConditions to run through these checks again for no additional code.